src/plugin.js

```diff
@@ -695,46 +710,71 @@ void main() {
     videoElStyle.zIndex = '-1';
     videoElStyle.opacity = '0';
 
+    let displays = [];
+    let hasWebXR = false;
+
     if (window.navigator.getVRDisplays) {
       this.log('is supported, getting vr displays');
-      window.navigator.getVRDisplays().then((displays) => {
-        if (displays.length > 0) {
-          this.log('Displays found', displays);
-          this.vrDisplay = displays[0];
-
-          // Native WebVR Head Mounted Displays (HMDs) like the HTC Vive
-          // also need the cardboard button to enter fully immersive mode
-          // so, we want to add the button if we're not polyfilled.
-          if (!this.vrDisplay.isPolyfilled) {
-            this.log('Real HMD found using VRControls', this.vrDisplay);
-            this.addCardboardButton_();
-
-            // We use VRControls here since we are working with an HMD
-            // and we only want orientation controls.
-            this.controls3d = new VRControls(this.camera);
-          }
-        }
 
-        if (!this.controls3d) {
-          this.log('no HMD found Using Orbit & Orientation Controls');
-          const options = {
-            camera: this.camera,
-            canvas: this.renderedCanvas,
-            // check if its a half sphere view projection
-            halfView: this.currentProjection_.indexOf('180') === 0,
-            orientation: videojs.browser.IS_IOS || videojs.browser.IS_ANDROID || false
-          };
-
-          if (this.options_.motionControls === false) {
-            options.orientation = false;
-          }
+      window.navigator.getVRDisplays().then((displaysArray) => {
+        displays = displaysArray;
+      });
+    }
 
-          this.controls3d = new OrbitOrientationContols(options);
-          this.canvasPlayerControls = new CanvasPlayerControls(this.player_, this.renderedCanvas, this.options_);
+    // Detect WebXR is supported
+    if (window.navigator.xr) {
+      this.log('WebXR is supported');
+      window.navigator.xr.isSessionSupported('immersive-vr').then((supportsImmersiveVR) => {
+        if (supportsImmersiveVR) {
```

Comment:
if this is `false` should we try and fall back to the `WebVRPolyfill`? Or does it happen automatically? A bit hard to understand since it's asynchronous.

Reply:
Yeah I went through a few permutations of that - after testing it seemed to not really matter
Then again could be getting lucky in timing each time